### PR TITLE
Added extent property to project and dataset. 

### PR DESCRIPTION
**Pull request by tony-gaia**
For project this is the extent of its geometry while for dataset it's the aggregate extent of all the records of that dataset.

### CODE REVIEW

File: biosys/apps/main/models.py [hunk 1]
```diff
@@ -193,6 +194,10 @@ def has_object_destroy_permission(self, request):
     def centroid(self):
         return self.geometry.centroid if self.geometry else None
 
+    @property
+    def extent(self):
+        return self.geometry.extent if self.geometry else None
```
Comment (serge-gaia) [inline biosys/apps/main/models.py:199]: Question: how `None` is represented in a `serializers.ListField`? Empty list or None?

File: biosys/apps/main/models.py [hunk 2]
```diff
@@ -342,7 +347,14 @@ def record_queryset(self):
 
     @property
     def record_count(self):
-        return self.record_model.objects.filter(dataset=self).count()
+        return self.record_queryset.count()
+
+    @property
+    def extent(self):
+        if self.type != Dataset.TYPE_GENERIC:
```
Comment (serge-gaia) [inline biosys/apps/main/models.py]: in prevision of the  removal of the dataset type you should remove this test and add a filter in the queryset
```
self.record_queryset.exclude(geometry__isnull=True).....
```
or maybe the Extent function does it

**Review by serge-gaia — COMMENTED** [review]
Just remove the test for generic and you can merge.

File: biosys/apps/main/models.py (hunk 1 above)
Comment (tony-gaia) [reply biosys/apps/main/models.py:199]: Comes out None

**Pull request merged**